### ISSUE TITLE
mingw32, mingw32-w64, libpython and cython

### ISSUE THREAD

**Issue by aldanor** [ContinuumIO/anaconda-issues#158]
I've spent some time lately trying to build some Cython extensions on Windows, and here's some of my observations:

1) `mingw` package that you provide is essentially a `mingw32-w64` which is quite useless for working with Cython (ref: https://github.com/cython/cython/wiki/64BitCythonExtensionsOnWindows). Correct me if I'm wrong. Try building `msgpack-python` with it and then run the tests to see what I mean (all tests pass just fine with plain vanilla mingw32; I had to build libpython.a for mingw32 for all Python interpreters first but that's quite simple, I can share a recipe if you want).

2) `mingw` package is obviously a stripped down version of mingw-w64, however since the package is already 250MB, it would make sense to include all possible mingw packages and dev tools like libtoolize, various autoconf stuff etc.

3) `libpython` has `mingw` as a dependency, which doesn't make a lot of sense if one wants to use an external mingw installation (which is reasonable given the size of `mingw` package).

TL;DR please provide mingw32 toolchain and libpython-mingw32 to let users build cython stuff; optionally, also remove mingw from libpython dependencies to allow for external mingw installation

Thanks.

@asmeurer, @ilanschnell  ^
